lib/auto-languageclient.ts

```diff
@@ -263,6 +274,7 @@ export default class AutoLanguageClient {
       (filepath) => this.filterChangeWatchedFiles(filepath),
       this.reportBusyWhile,
       this.getServerName(),
+      (e) => this.determineProjectPath(e),
```

Comment:
Instead of all these, we can probably provide a method that is passed down to `this._serverManager`.

Passed as an argument:
```js
      this.customDetermineProject
```

Later we will check if this method is defined to use that instead.